gnomad_qc/v4/assessment/calculate_per_sample_stats.py

```diff
@@ -303,50 +352,62 @@ def main(args):
+                mt = get_gnomad_v4_genomes_vds(
+                    test=test, release_only=True, split=True, chrom=chrom
+                ).variant_data
+
+            release_ht = release_sites(data_type=data_type).ht()
+            if test:
+                release_ht = hl.filter_intervals(
+                    release_ht, [hl.parse_locus_interval("chr22")]
+                )
 
-        create_per_sample_counts_ht(
-            mt,
-            release_ht,
-            pass_filters=not args.skip_pass_filters,
-            ukb_capture=not args.skip_filter_ukb_capture_intervals,
-            broad_capture=not args.skip_filter_broad_capture_intervals,
-            by_csqs=not args.skip_by_csqs,
-            rare_variants=not args.skip_rare_variants,
-            vep_canonical=args.vep_canonical,
-            vep_mane=args.vep_mane,
-            rare_variants_af=args.rare_variants_af,
-        ).write(per_sample_res.path, overwrite=overwrite)
-
-    if args.aggregate_sample_stats:
-        logger.info("Computing aggregate sample statistics...")
-        ht = per_sample_res.ht().checkpoint(
-            hl.utils.new_temp_file("per_sample_counts", "ht")
-        )
-        ht = compute_agg_sample_stats(
-            ht,
-            meta(data_type=data_type).ht(),
-            by_ancestry=args.by_ancestry,
-            by_subset=args.by_subset,
-        )
-        ht.write(per_sample_agg_res.path, overwrite=overwrite)
+            create_per_sample_counts_ht(
+                mt,
+                release_ht,
+                pass_filters=not args.skip_pass_filters,
+                ukb_capture=(
+                    not args.skip_filter_ukb_capture_intervals
+                    if data_type == "exomes"
+                    else False
+                ),
+                broad_capture=(
+                    not args.skip_filter_broad_capture_intervals
+                    if data_type == "exomes"
+                    else False
+                ),
+                by_csqs=not args.skip_by_csqs,
+                rare_variants=not args.skip_rare_variants,
+                vep_canonical=args.vep_canonical,
+                vep_mane=args.vep_mane,
+                rare_variants_afs=args.rare_variants_afs,
+            ).write(per_sample_res.path, overwrite=overwrite)
+
+        if args.aggregate_sample_stats:
+            logger.info("Computing aggregate sample statistics...")
+            ht = per_sample_res.ht().checkpoint(
+                hl.utils.new_temp_file("per_sample_counts", "ht")
+            )
+            ht = compute_agg_sample_stats(
+                ht,
+                meta(data_type=data_type).ht(),
+                by_ancestry=args.by_ancestry,
+                by_subset=args.by_subset,
+            )
+            ht.write(per_sample_agg_res.path, overwrite=overwrite)
+    finally:
+        logger.info("Copying log to logging bucket...")
+        hl.copy_log(get_logging_path("per_sample_stats"))
```

Comment:
I'm not super familiar with the 'try:' 'finally:' logic, but that's what Kristen uses in some mito scripts and I've seen it elsewhere in the repo. Might have to leave this to Julia to have anything smarter to say

Reply:
This is for the log of a possible exception error, we think we might run into this while running on the whole dataset.

Reply:
good with me!